controller/api/destination/watcher/ip_watcher.go

```diff
@@ -93,6 +95,18 @@ func NewIPWatcher(k8sAPI *k8s.API, endpoints *EndpointsWatcher, log *logging.Ent
 		},
 	})
 
+	k8sAPI.Pod().Informer().AddIndexers(cache.Indexers{hostIPIndex: func(obj interface{}) ([]string, error) {
+		if pod, ok := obj.(*corev1.Pod); ok {
+			// If the pod does not have a host IP it should not be added to
+			// the host IP indexer.
+			if pod.Status.HostIP != "" {
+				return []string{pod.Status.HostIP}, nil
```

Comment:
It seems like here in the indexer, we have enough information to index by port as well? i.e. here we could return a list of HostIP:HostPort values such that GetPod can do a single lookup rather than having to iterate through all pods on the host?

Reply:
Oh that's a good idea! I'll make that change.